components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -1396,15 +1397,24 @@ private Cookie getTokenIdCookie(HttpServletRequest req) {
      * @param sessionId            Session Id.
      * @param req                  HttpServlet Request.
      * @param resp                 HttpServlet Response.
+     * @param tenantDomain         Tenant Domain
      * @param loggedInTenantDomain Logged In Tenant Domain.
+     * @param sessionIdentifier    Session Identifier
      */
     private void storeTokenIdCookie(String sessionId, HttpServletRequest req, HttpServletResponse resp,
-                                    String tenantDomain, String loggedInTenantDomain) {
+                                    String tenantDomain, String loggedInTenantDomain, String sessionIdentifier) {
```

Comment:
What is the difference of `sessionId` and `sessionIdentifier` params?

Reply:
sessionId is the cookie value of samlssoTokenId cookie (Could be a random UUID as well). sessionIdentifier is used to uniquely identify the session. Those values are different from one another.